notebook/static/tree/js/notebooklist.js

```diff
@@ -697,6 +707,73 @@ define([
         });
     };
 
+    NotebookList.prototype.move_selected = function() {
+        // TODO(nhdaly): Support moving multiple items at once.
+        if (this.selected.length !== 1){
+            return;
+        }
+
+        var that = this;
```

Comment:
Good question, @NHDaly! I'm gonna give explaining it a shot and let me know if I am unclear. It all boils down to scoping. In JavaScript, the `this` keyword is used to refer to the object that the function you are currently in is tied to. In this case, `this` references `NotebookList`. You can confirm this by adding a `console.log(this)` statement to the beginning of the `move_selected` function. Now, let's say that inside your `move_selected` function, you create a new object with functions as you do with the dialog in this case. Your `this` now references a completely different object within those functions. You can test this out by adding a `console.log(this)` right before line 741. But what if in this new context you want to reference the `NotebookList` as you do on line 741? Well, you store the value of `this` in a variable called `that` when `this` references the object you later want to reference and then use `that` in functions where `this` does not reference that object.

So in reality, you don't particularly need it for the first few lines at the top since `this` references `NotebookList` in those cases but I like to declare at the top and use it consistently so that it is obvious that throughout the code anywhere `that` is used we are referencing a `NotebookList` object.

Let me know if any of that needs clarification! `this` is definitely a tricky topic in JavaScript.